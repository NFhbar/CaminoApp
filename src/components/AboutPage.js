import React from 'react';
import { NavLink } from 'react-router-dom';
import { row, column } from 'reactstrap';

const AboutPage = () => (
  <div className="row">
    <div className="col-sm-12">
      <p>From About</p>
    </div>
  </div>
);

export default AboutPage;
