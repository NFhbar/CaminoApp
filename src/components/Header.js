import React from 'react';
import { NavLink } from 'react-router-dom';
import { row, column } from 'reactstrap';

const Header = () => (
  <div className="row">
    <div className="col-sm-12">
      <h1>CaminoApp</h1>
      <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
      <NavLink to="/create" activeClassName="is-active">Create Item</NavLink>
      <NavLink to="/about" activeClassName="is-active">About</NavLink>
    </div>
  </div>
);

export default Header;
