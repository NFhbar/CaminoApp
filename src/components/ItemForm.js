import React from 'react';
import { alert, button } from 'reactstrap';

export default class ItemForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: props.item ? props.item.description : '',
      note: props.item ? props.item.note : '',
      amount: props.item ? (props.item.amount /100).toString() : '',
      error: ''
    };
  }
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };
  onAmountChange = (e) => {
    const amount = e.target.value;

    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      this.setState(() => ({ error: 'Please provide and item and amount.'}));
    } else {
      this.setState(() =>({ error: ''}));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10)*100,
        note: this.state.note
      });
    }
  };
  render() {
    return (
      <div>
        {this.state.error &&
          <div className=
            "alert
            alert-warning
            alert-dismissible
            fade
            show"
            role="alert">
            {this.state.error}
          </div>
        }
        <form onSubmit={this.onSubmit}>
          <div className="form-group">
            <input
              type="text"
              placeholder="Description"
              autoFocus
              value={this.state.description}
              onChange={this.onDescriptionChange}
              className="form-control"
            />
          </div>
          <div className="form-group">
            <input
              type="text"
              placeholder="Amount"
              value={this.state.amount}
              onChange={this.onAmountChange}
              className="form-control"
              />
          </div>
          <div className="form-group">
            <textarea
              placeholder="Add a note for your item (optional)"
              value={this.state.note}
              onChange={this.onNoteChange}
              className="form-control"
              >
            </textarea>
          </div>
          <button type="submit" className="btn btn-lg btn-primary">Submit Item</button>
        </form>
      </div>
    )
  }
}
