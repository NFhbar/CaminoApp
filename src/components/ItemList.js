import React from 'react';
import { connect } from 'react-redux';
import ItemListItem from './ItemListItem';
import selectItems from '../selectors/items';

export const ItemList = (props) => (
  <div>
    {
      props.items.length === 0 ? (
        <p>No Items</p>
      ) : (
        props.items.map((item) => {
          return <ItemListItem key={item.id} {...item}/>
        })
      )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    items: selectItems(state.items, state.filters)
  };
};

export default connect(mapStateToProps)(ItemList);
