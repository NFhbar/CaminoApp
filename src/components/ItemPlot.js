import React from 'react';
import { connect } from 'react-redux';
import ItemPlotItem from './ItemPlotItem';
import selectPlotData from '../selectors/plot-data';

export const ItemPlot = (props) => {
    if(props.plots.length === 0) {
      return(
        <div className="alert alert-primary" role="alert">Add some items to plot sales!</div>
      )
    } else {
      return (
        <ItemPlotItem plots={props.plots}/>
      )
    }
};

const mapStateToProps = (state) => {
  return {
    plots: selectPlotData(state.items, state.filters)
  };
};

export default connect(mapStateToProps)(ItemPlot);
