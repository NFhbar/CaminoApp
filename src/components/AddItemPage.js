import React from 'react';
import { connect } from 'react-redux';
import { row, column } from 'reactstrap';
import ItemForm from './ItemForm'
import { addItem } from '../actions/items';

export class AddItemPage extends React.Component {
  onSubmit = (item) => {
    this.props.addItem(item);
    this.props.history.push('/');
  };
  render() {
    return (
      <div className="row card-body">
        <div className="col-sm-12">
          <h3>Add an item to your store!</h3>
          <ItemForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addItem: (item) => dispatch(addItem(item))
});

export default connect(undefined, mapDispatchToProps)(AddItemPage);
