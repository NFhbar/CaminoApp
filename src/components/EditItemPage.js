import React from 'react';
import { connect } from 'react-redux';
import { row, column } from 'reactstrap';
import ItemForm from './ItemForm';
import { editItem, startRemoveItem } from '../actions/items';

export class EditItemPage extends React.Component {
  onSubmit = (item) => {
    this.props.editItem(this.props.item.id, item);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveItem({ id: this.props.item.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div className="row" id="edit-item-page">
        <div className="col-sm-12">
          <ItemForm
            item={this.props.item}
            onSubmit={this.onSubmit}
          />
          <button type="submit" id="btn-remove" className="btn btn-md btn-danger" onClick={this.onRemove}>Remove Item</button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  item: state.items.find((item) => item.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  editItem: (id, item) => dispatch(editItem(id, item)),
  startRemoveItem: (data) => dispatch(startRemoveItem(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditItemPage);
