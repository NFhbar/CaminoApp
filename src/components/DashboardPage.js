import React from 'react';
import { row, column } from 'reactstrap';
import ItemList from './ItemList';
import ItemPlot from './ItemPlot';
import ItemListFilters from './ItemListFilters';

const DashboardPage = () => (
  <div className="row card-body">
    <div className="col-sm-12">
      <ItemListFilters />
      <ul className="list-group">
        <li className="list-group-item disabled">
          <div className="row">
            <div className="col-sm-4">
              Item
            </div>
            <div className="col-sm-4">
              Price
            </div>
            <div className="col-sm-4">
              Items sold
            </div>
          </div>
        </li>
        <ItemList />
      </ul>
      <ItemPlot />
    </div>
  </div>
);

export default DashboardPage;
