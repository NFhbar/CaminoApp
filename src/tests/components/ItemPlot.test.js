import React from 'react';
import { shallow } from 'enzyme';
import {ItemPlot} from '../../components/ItemPlot';
import plotData from '../fixtures/plot-data';

test('should render ItemPlot with plot data', () => {
  const wrapper = shallow(<ItemPlot plots={plotData} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ItemPlot with empty message', () => {
  const wrapper = shallow(<ItemPlot plots={[]} />);
  expect(wrapper).toMatchSnapshot();
});
