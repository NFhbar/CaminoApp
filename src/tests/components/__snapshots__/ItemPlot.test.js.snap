// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ItemPlot with empty message 1`] = `
<div
  className="alert alert-primary"
  role="alert"
>
  Add some items to plot sales!
</div>
`;

exports[`should render ItemPlot with plot data 1`] = `
<ItemPlotItem
  plots={
    Array [
      Object {
        "description": "Pant",
        "sales": 10,
        "sales2": 20,
        "sales3": 30,
        "sales4": 40,
      },
      Object {
        "description": "Shirt",
        "sales": 1,
        "sales2": 2,
        "sales3": 3,
        "sales4": 4,
      },
      Object {
        "description": "Coat",
        "sales": 5,
        "sales2": 10,
        "sales3": 15,
        "sales4": 20,
      },
    ]
  }
/>
`;
